### PR TITLE
Bare Metal: Document maxSurge and maxUnvailable configuration parameter for rolling upgrade

### PR DESCRIPTION
**Pull request by ptrivedi**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md [hunk 1]
```diff
@@ -203,6 +203,108 @@ kubectl annotate clusters.anywhere.eks.amazonaws.com ${CLUSTER_NAME} -n ${CLUSTE
...
+This field accepts configuration parameters for customizing rolling upgrade behavior.
+
+#### maxSurge
+Default: 1
+
+This can not be 0 if maxUnavailable is 0.
+
+The maximum number of machines that can be scheduled above the desired number of machines. 
+
+Example: When this is set to n, the new worker node group can be scaled up immediately by n when the rolling update starts. Total number of machines in the cluster (old + new) never exceeds (desired number of machines + n). Once scale down happens and old machines are brought down, the new worker node group can be scaled up further ensuring that the total number of machines running at any time does not exceed the desired number of machines + n.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:289]: Change "rolling update" to "rolling upgrade"?

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md [hunk 2]
```diff
@@ -203,6 +203,108 @@ kubectl annotate clusters.anywhere.eks.amazonaws.com ${CLUSTER_NAME} -n ${CLUSTE
...
+
+#### maxUnavailable
+Default: 0
+
+This can not be 0 if MaxSurge is 0.
+
+The maximum number of machines that can be unavailable during the update.
+
+Example: When this is set to n, the old worker node group can be scaled down by n machines immediately when the rolling update starts. Once new machines are ready, old worker node group can be scaled down further, followed by scaling up the new worker node group, ensuring that the total number of machines unavailable at all times during the update never falls below n.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:298]: Change "rolling update" to "rolling upgrade"?

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md [hunk 3]
```diff
@@ -203,6 +203,108 @@ kubectl annotate clusters.anywhere.eks.amazonaws.com ${CLUSTER_NAME} -n ${CLUSTE
...
+
+### Rolling upgrades with no additional hardware
+
+When maxSurge is set to 0 and maxUnavailable is set to 1, it allows for a rolling upgrade without need for additional hardware. Use this configuration if your workloads can tolerate node unavailability.
+
+>**_NOTE:_** This could ONLY be used if unavailability of a maximum of 1 node is acceptable.
+
+With this kind of configuration, the rolling upgrade will proceed node by node, deprovision and delete a node fully before re-provisioning it with upgraded version and re-joining it to the cluster. This means that any point during the course of the rolling upgrade, there could be one unavailable node.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:306]:
```suggestion
With this kind of configuration, the rolling upgrade will proceed node by node, deprovision and delete a node fully before re-provisioning it with upgraded version, and re-join it to the cluster. This means that any point during the course of the rolling upgrade, there could be one unavailable node.
```

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md (hunk 1 above)
Comment (ptrivedi) [reply docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:289]: CAPI's 'strategy type' field calls it 'RollingUpdate', but agree that 'rolling upgrade' is clearer

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md [hunk 4]
```diff
@@ -203,6 +203,105 @@ kubectl annotate clusters.anywhere.eks.amazonaws.com ${CLUSTER_NAME} -n ${CLUSTE
 `Cluster`:
 - `kubernetesVersion`
 
+### Advanced configuration for rolling upgrade
+
+EKS Anywhere allows an optional configuration to customize the behavior of upgrades. 
+
+It allows the specification of 
+* Type of upgrade strategy (only 'RollingUpdate' is currently supported)
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:211]: Is `RollingUpdate` a literal value? I don't see anything called that.

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md [hunk 5]
```diff
@@ -203,6 +203,105 @@ kubectl annotate clusters.anywhere.eks.amazonaws.com ${CLUSTER_NAME} -n ${CLUSTE
...
+* Two parameters that control the desired behavior of rolling upgrades: 
+  * maxSurge - The maximum number of machines that can be scheduled above the desired number of machines. When not specified, the current CAPI default of 1 is used.
+  * maxUnavailable - The maximum number of machines that can be unavailable during the update. When not specified, the current CAPI default of 0 is used.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:214]: Change "update" to "upgrade"?

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md [hunk 6]
```diff
@@ -203,6 +203,105 @@ kubectl annotate clusters.anywhere.eks.amazonaws.com ${CLUSTER_NAME} -n ${CLUSTE
...
+The maximum number of machines that can be scheduled above the desired number of machines. 
+
+Example: When this is set to n, the new worker node group can be scaled up immediately by n when the rolling upgrade starts. Total number of machines in the cluster (old + new) never exceeds (desired number of machines + n). Once scale down happens and old machines are brought down, the new worker node group can be scaled up further ensuring that the total number of machines running at any time does not exceed the desired number of machines + n.
+
+#### maxUnavailable
+Default: 0
+
+This can not be 0 if MaxSurge is 0.
+
+The maximum number of machines that can be unavailable during the update.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:293]: Change "update" to "upgrade"?

File: docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md [hunk 7]
```diff
@@ -203,6 +203,105 @@ kubectl annotate clusters.anywhere.eks.amazonaws.com ${CLUSTER_NAME} -n ${CLUSTE
...
+
+Example: When this is set to n, the old worker node group can be scaled down by n machines immediately when the rolling upgrade starts. Once new machines are ready, old worker node group can be scaled down further, followed by scaling up the new worker node group, ensuring that the total number of machines unavailable at all times during the update never falls below n.
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades/baremetal-upgrades.md:295]: Change "update" to "upgrade"?

**Pull request merged**